### ISSUE TITLE
[A11y] input type="number" offers rocker-buttons that are not accessible

### ISSUE THREAD

**Issue by wiffbi** [pretix/pretix#2211]
**Describe the bug**
I myself do not see this as a bug, but it came up during a11y-testing. Those rocker-buttons are not accessible by keyboard and might impose a barrier to sighted users with cognitive constraints as these buttons do not offer any info regarding their functionality.

**Expected behavior**
Unknown. Remove rocker-buttons altogether and change to `<input type="text" inputmode="numeric">`?

**Comment by wiffbi** [comment]
Another alternative as noted in #1128 might be to use `<select>` with options from 1-10 (or maybe some clever logic to limit max to quota if it is less).
![Bildschirmfoto 2021-10-01 um 12 42 51](https://user-images.githubusercontent.com/276509/135608948-7299ae10-8b5f-48ef-9e57-de4c0f610413.png)
![Bildschirmfoto 2021-10-01 um 12 45 31](https://user-images.githubusercontent.com/276509/135608960-25b0c29d-2fb0-4ef8-b281-01af5c094766.png)
![Bildschirmfoto 2021-10-01 um 12 47 19](https://user-images.githubusercontent.com/276509/135608969-04971169-e52d-452c-b284-32521a97390d.png)

A different idea might be to add a table-head like label for the column. I am not sure we should markup the whole section as a table, but for sighted users this might be an option.

![pretix-products-table](https://user-images.githubusercontent.com/276509/135609114-0a4c695e-e359-472a-b571-cd16612ed66f.png)

**Comment by wiffbi** [comment]
As @raphaelm pointed out, adding more than 10 or even up to 100 products to the cart (not uncommon for certain types of ticket shops) does not scale with a `<select>`. As an alternative I suggest, we try a combination of both: start with a select-element with an added option „more …“ at the end, which when selected turns the select into a number-input (or to be more precise: text-input with inputmode=numeric to pass a11y-tests).

Having a select-element with the first/empty option reading „Quantity“ (or even „Quantity: 0“) could improve usability by adding a visual „label“ to tell the user, what this is input is about. Furthermore it solves this very issue (accessibility of rocker-buttons).

**Comment by raphaelm** [comment]
> A different idea might be to add a table-head like label for the column. I am not sure we should markup the whole section as a table, but for sighted users this might be an option.

Unfortunately this is the only of your screenshots where I don't think it looks ugly/crowded, but it's also the least accessible one.

> As @raphaelm pointed out, adding more than 10 or even up to 100 products to the cart (not uncommon for certain types of ticket shops) does not scale with a <select>. As an alternative I suggest, we try a combination of both: start with a select-element with an added option „more …“ at the end, which when selected turns the select into a number-input (or to be more precise: text-input with inputmode=numeric to pass a11y-tests).

That would mean mandatory JavaScript for buying tickets, which we currently avoid, wouldn't it?

**Comment by wiffbi** [comment]
> > A different idea might be to add a table-head like label for the column. I am not sure we should markup the whole section as a table, but for sighted users this might be an option.
> Unfortunately this is the only of your screenshots where I don't think it looks ugly/crowded, but it's also the least accessible one.

Regarding _uglyness:_ thanks. :) Yes, it actually looks better than the rest. Regarding accessibility – we keep the screen reader `<label>`s and add these just for sighted users, so it should be better than before. Only thing is, that this does not translate to mobile at all, so we would need to hide them there. :(

> That would mean mandatory JavaScript for buying tickets, which we currently avoid, wouldn't it?

Well, it depends on how its implemented. If we start with `<input type="text" inputmode="numeric">` as a baseline in HTML and dynamically replace it with `<select>` with JavaScript, then everything works as before even without JavaScript. If we start with `<select>` in the HTML, then at least for up to 10 tickets it works without JS. Of course I’d prefer the first option similar to progressive enhancement. 

Of course, all of this is only necessary, if we agree, that having a select-field is easier to understand for the user what to do with the input field (I am not so sure about, maybe with the added first `<option>Quantity: 0</option>`). And of course how important we think it is to remove these rocker-buttons to pass a11y-testing (regarding certification they interpret it as a blocking barrier :().

I will think about it some more. Maybe you are right as mentioned in #1128 that we need a completely different approach even design-wise. A11y-Testers did not list as blocking point, but during our meeting told me, that having the input so far away from the product’s label/name is not a good idea. I will look at my design-ideas from back then, although being out of scope for this PR, maybe I find some ideas in there.

**Comment by wiffbi** [comment]
I looked through some old design ideas and found these. I like that they have a visual label, tell the min (price) and max (quantity) if available. Putting the total available under the quantity input kind of makes sense, but combined with a max quantity per order might be a bit confusing. But showing total available anywhere with a max quantity per order might be confusing to the user, so probably not much to do about it … 

![Bildschirmfoto 2021-10-05 um 08 09 59](https://user-images.githubusercontent.com/276509/135969755-2831b2e9-2aa6-443d-afb7-9763011d987a.png)

The only thing is, they probably would not work in the current design as they are to „heavy“ – especially if products only have a title and nothing else. Then on the left only one line of text (the product title) and on the right these „heavy“ price and quantity inputs would not look good.

Just for completeness, this is where I picked these elements from (the other +- buttons are the wrong color combination):

![Bildschirmfoto 2021-10-05 um 08 15 24](https://user-images.githubusercontent.com/276509/135970405-a6559c4a-0a3a-494d-925c-b30b979150ba.png)


Sooo, what do we do with this? What if I just take the number-input element for quantity and swap it with this one? This will at least fix the rocker-buttons accessibility issue. Not sure why, but I have the feeling due to these +- buttons I expect them to be added to the cart immediately instead of having to hit the „add to cart“ button. With a native select, IMHO I expect to hit a submit-button somewhere. But this is once again probably just me.

![Bildschirmfoto 2021-10-05 um 08 21 05](https://user-images.githubusercontent.com/276509/135970890-766b571a-7d78-4ecf-a463-5189ad833c7e.png)

**Comment by wiffbi** [comment]
P.S.: Also, visually, the rocker input is very close to the quantity inputs in the cart, which actually work like what I described my feeling with these rocker-buttons is: they immediately change the quantity in the cart.

![Bildschirmfoto 2021-10-05 um 08 33 25](https://user-images.githubusercontent.com/276509/135972236-6eb2bf04-677f-402e-8daa-73ac927192ab.png)

This might be to confusing, I guess. Two elements, that do the same (increase/decrease quantity) looking only slightly different (basically just a styling thing), but working totally different.

Back to the drawing board.
